interface/main/calendar/add_edit_event.php

```diff
@@ -1072,6 +1087,15 @@ function sel_patient() {
     dlgopen('find_patient_popup.php', 'findPatient', 650, 300, '', title);
 }
 
+// This invokes the cancellation popup.
+<?php
+if (stristr($row['pc_title'], 'telehealth') && ismoduleactive() >= 1) {
+    $sendid = new AppointmentAddEvent();
+    $sendid->setEventid($eid);
+    $eventDispatcher->dispatch(AppointmentAddEvent::ACTION_RENDER_CANCEL_JAVASCRIPT, new GenericEvent());
```

Comment:
@juggernautsei Wouldn't you want to send the $sendid object here instead of GenericEvent?

Reply:
When I tried that it did not work for me. I did this

$eventDispatcher->dispatch(AppointmentAddEvent::ACTION_RENDER_CANCEL_JAVASCRIPT, new AppoingmentAddEvent($eid));

Here is the error I got.
PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function OpenEMR\\Events\\Appointments\\AppointmentAddEvent::__construct(), 0 passed in /var/www/html/mindful_v6/interface/modules/custom_modules/oe-module-lifemesh-telehealth/openemr.bootstrap.php on line 35 and exactly 1 expected in /var/www/html/mindful_v6/src/Events/Appointments/AppointmentAddEvent.php:31\nStack trace:\n#0 /var/www/html/mindful_v6/interface/modules/custom_modules/oe-module-lifemesh-telehealth/openemr.bootstrap.php(35): OpenEMR\\Events\\Appointments\\AppointmentAddEvent->__construct()\n#1 /var/www/html/mindful_v6/vendor/symfony/event-dispatcher/EventDispatcher.php(264): oe_module_lifemesh_telehealth_cancel_javascript()\n#2 /var/www/html/mindful_v6/vendor/symfony/event-dispatcher/EventDispatcher.php(239): Symfony\\Component\\EventDispatcher\\EventDispatcher->doDispatch()\n#3 /var/www/html/mindful_v6/vendor/symfony/event-dispatcher/EventDispatcher.php(73): Symfony\\Component\\EventDispatcher\\EventDispatcher->callListeners()\n#4 /var/www/html/mindful_v6/interf in /var/www/html/mindful_v6/src/Events/Appointments/AppointmentAddEvent.php on line 31, referer: https://dev.affordablecustomehr.com/mindful_v6/interface/main/tabs/main.php?token_main=URyEL4zqwEaw6vZKFMrXZGwXiWh1EvJpm7zh2TEZ
[Fri Sep 10 18:53:51.869966 2021] [php7:notice] [pid 488430] [client 70.184.171.187:54957] PHP Notice:  Undefined index: group in /var/www/html/mindful_v6/interface/main/calendar/add_edit_event.php on line 133, referer:

Reply:
That's really odd, so for some reason it wasn't recognizing the $eid?  Sorry for the delayed responses, I've been doing quite a bit of traveling the last few weeks.